### ISSUE TITLE
Transfering results of Async Requests to Queue Best Practice

### ISSUE THREAD

**Issue by avickars** [triton-inference-server/server#4872]
**Description**

**Triton Information**
What version of Triton are you using?
22.06

Are you using the Triton container or did you build it yourself?
Container

**To Reproduce**
Steps to reproduce the behavior.

Take "tc::InferResult* result" and put it on a queue for another process to pick up (obviously the queue is protected).  Almost immediately get a seg fault.

Describe the models (framework, inputs, outputs), ideally include the model configuration file (if using an ensemble include the model configuration file for that as well).

**Expected behavior**

Expected result is to be able to "tc::InferResult* result" on a queue for another process to consume it.

Using Tritons c++ client API, we are creating a pipeline of sorts where we would like take the inference result (during the callback) of an async request and put it on a queue for another process to consume it.  Unfortunately we are experience a seg fault, that Im guessing is likely as a result of the result potentially being destroyed once the callback finished, however it is unclear.

We are working off of async client (https://github.com/triton-inference-server/client/blob/main/src/c%2B%2B/examples/simple_grpc_async_infer_client.cc), and an specifically referencing this part here (https://github.com/triton-inference-server/client/blob/3d05400ed8962c03cb38d45ac4e61276950c3a7a/src/c%2B%2B/examples/simple_grpc_async_infer_client.cc#L265).

Basically, we are hoping to understand what the best practice is for doing this (i.e. how exactly to keep the inference result from being destroyed).  We are essentially looking to do what is done in the python client async example where the result is added to a list.

I can of course see that you guys are creating a shared pointer to hold the result, however Im not understanding why since Im assuming that the result wouldn't be destroyed for the duration of the callback?

Any help is greatly appreciated!

Thanks,

Aidan

**Comment by avickars** [comment]
figured it out. Im an idiot.
